### PR TITLE
KB for checking lab units and developer notes

### PR DESCRIPTION
**Pull request by vojtechhuser**
Submission of knowledge base and developer notes to extend DQ checking for measurements.

See context here
https://github.com/OHDSI/StudyProtocolSandbox/blob/master/themis/extras/notes.md#update-2018-11

and in that repository for more about ThemisMeasurement.
(just presented at AMIA)

realted issues #351  #250

### CODE REVIEW

**Comment by vojtechhuser** [comment]
@alondhe  can you please review. I want to be sure that if I work on next steps, it will not hang in the PR status for months. So this is a true test of the team vote on using CSV knowledge base.

My next step is to do follow up PR with checkThemis R function

**Comment by t-abdul-basser** [comment]
I responded with a question in my review on the same day that this PR was posted. I just reposted it above, since it appears that it was not seen.  

To elaborate on my review comments: I do not understand the purpose of this PR. Why was it submitted without the code that apparently should accompany it? If it is meant to be a proposal, it should be an issue, rather than a PR?

On a related note, at Achilles' currently release cadence, contribution frequency and active contributor size, the average PR review and closure rate is quite reasonable. It also does not approach a month (as a review of closed PRs and project statistics shows). We need to improve the stability of the code base and merging RP without a thorough review seems ill-advised.

File: README-developers.md [hunk 1]
```diff
@@ -197,3 +197,7 @@ Follow the conventions of existing Heel Results queries:  `select <fields> into
 
 * The `@hrOldId` parameter refers to the serial file ID of the previous achilles_heel_results analysis. The achillesHeel function will assign this based on the rule_id.
 
+
```
[the remark this answers is not in the document]
Comment (vojtechhuser) [reply README-developers.md]: The code is in DataQuality package. I need buy-in from you and Ajit first. Looks like you oppose or I am not sure what is wrong with my PR.

**Comment by alondhe** [comment]
Sorry @vojtechhuser I was having trouble parsing the diff in Git (I'm still learning how to interpret certain GitHub pages). I have no opposition, just confirming the changes:

1. adding the blurb to the developer readme
2. adding the themis rules CSV

That's it, right? If so, I will hit approve.

**Comment by chrisknoll** [comment]
I think @vojtechhuser  has this backwards:  we don't need to set up an additional dependency on themis in the achilles repo.  The set of 'network results' should reside in the themis measurements study, and instructions on how to incorporate the achilles output for analysis with the themis results can be described _there_.  Otherwise, every time a themis update happens (which I don't believe has the same release cadence as Achilles, if any) we'll have to do some sort of Achilles release...a sort of 'tail wagging the dog'.  If themis wants to apply some methodology against some output of an Achilles Version, that's simple enough to declare dependency of themis to Achilles.  If you can imagine that if 5 or more different parties had a different way of leveraging the achilles results, and that resulted in 5 different parties demanding their own release schedule of Achilles, you can see where this would get out of hand.  

So, I recommend putting the themis assets in themis, and providing instructions in the themis repo about how to leverage Achilles results to perform some sort of method for analysis.

**Comment by t-abdul-basser** [comment]
I am incline to agree with @chrisknoll.

**Comment by vojtechhuser** [comment]
@alondhe Yes. It is right. I am glad at least one developer is in favor.

**Comment by vojtechhuser** [comment]
@chrisknoll I am proposing a static KB that is updated once a year. (after 3+ plus years maybe never)

 There is no super formal Themis repo with code. Achilles should not depend on a sandbox repository.

I will file an issue as well for the developer note piece.

**Comment by chrisknoll** [comment]
@t-abdul-basser , @alondhe , I don't want to overstep my bounds here since you both have been primary custodians of the Achilles repository, and if anything I expressed above is not inline with your vision or planning, I apologize for my misunderstanding.

**Comment by t-abdul-basser** [comment]
@chrisknoll I appreciate that. To be clear, I believe that we can handle DQ in (or through) Achilles. I appreciate @vojtechhuser efforts to link THEMIS with Achilles and think that doing so is consistent with Achilles mission and evolution. I would suggest however that the way in which we do that should be broadly consistent with what @vojtechhuser @chrisknoll @alondhe @cgreich discussed earlier about R-, python-based Achilles add-on: namely that they be modularized and externalized.  
Accordingly, we could create a 1) repo for an Achilles module that conatins R code for checking data validity/quality based on THEMIS knowledge base/rules set and 2) modify Achilles 2.0 to support pluggable modules such as this first one.

**Comment by vojtechhuser** [comment]
the repo for 1) is here: https://github.com/OHDSI/StudyProtocolSandbox/blob/master/DataQuality/R/checkThemis.R 

Well, I have ajit voting for approving my PR.
>That's it, right? If so, I will hit approve.

I am formally re-submitting this contribution to Achilles. Taha, please make your vote.

(given that your point 2) is not there and I am not volunteering to do it. (did you have a developer in mind for 2); otherwise my PR will be waiting for that for a possibly a long time)

**Comment by t-abdul-basser** [comment]
* Regarding an R module to accompany the PR, thanks for the link to the `checkThemis` module.
* Regarding modifying Achilles to support plug-ins, I was not expecting you to implement that. Depending on what we decide regarding the roadmap for v2.0, I might implement the first version of the framework.

**Pull request merged**